.github/workflows/pr-build.yml

```diff
@@ -0,0 +1,48 @@
+name: PR Build
+
+on:
+  pull_request:
```

Comment:
We could use `pull_request_target`, and build and deploy in a single workflow but that's equivalent to "Hey, feel free to steal secrets :)".